ring/ring.go

```diff
@@ -866,16 +866,32 @@ func mergeTokenGroups(groupsByName map[string][]uint32) []uint32 {
 	return merged
 }
 
-// GetInstanceState returns the current state of an instance or an error if the
-// instance does not exist in the ring.
-func (r *Ring) GetInstanceState(instanceID string) (InstanceState, error) {
+// UnsafeGetInstance return the InstanceDesc for the given instanceID or an error
+// if the instance doesn't exist in the ring. The returned InstanceDesc is NOT a
+// deep copy, so the caller should never modify it.
+func (r *Ring) UnsafeGetInstance(instanceID string) (InstanceDesc, error) {
```

Comment:
There's nothing unsafe about *calling* this method. Returned InstanceDesc is a shallow-copy, and all of its attributes except tokens can also be freely changed. I disagree with using "Unsafe" prefix here (we don't use this anywhere in dskit), and think that enhanced comment is enough.

Perhaps naming return value as `(doNotModify InstanceDesc, _ error)` would also indicate what you want to say here.

Reply:
> and all of its attributes except tokens can also be freely changed

Reason why I called it unsafe was exactly because of the tokens.

Reply:
Addressed in https://github.com/grafana/dskit/pull/479/commits/e33bb87dc4c8eb13e91aeff9d22a8f35a4246950